### ISSUE TITLE
Removing large amounts of games causes memory exhaustion/crash

### ISSUE THREAD

**Issue by mavericm1** [doZennn/steam-rom-manager#563]
Originally working with the classic rom manager i was used to however with new emudeck i didn't realize i couldn't just save a certain console using the filtering options at the top of the preview which managed to add all 12k+ roms to steam.   The issue is i'm now struggling to get these entries removed and clean up my steam library.   And just use the parsers i need.   

top - 13:19:10 up 46 min,  4 users,  load average: 66.96, 55.70, 32.72
Tasks: 303 total,   2 running, 301 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.2 us,  9.1 sy,  0.0 ni,  0.6 id, 89.7 wa,  0.2 hi,  0.2 si,  0.0 st
MiB Mem :  14818.6 total,    132.5 free,  14439.5 used,    246.6 buff/cache
MiB Swap:   1024.0 total,      0.0 free,   1024.0 used.     14.7 avail Mem

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  13968 deck      20   0 1132.9g  12.6g   1088 D   0.1  87.2   5:05.51 steam-rom-manag
   3287 deck      20   0 3933144 284068     56 S   1.1   1.9   1:18.04 plasmashell
   3155 deck      20   0  845396  29016   9536 D   1.2   0.2   2:51.25 Xorg
  10667 deck      20   0   24.5g  26856   4744 D   0.0   0.2   0:55.63 emudeck
   3252 deck      20   0 1353788  22880     12 D   0.9   0.2   2:16.38 kwin_x11
  13948 deck      20   0   32.6g  19944    416 D   0.0   0.1   0:08.67 steam-rom-manag
   1800 deck      20   0  182048  16828     76 S   0.0   0.1   0:01.45 pipewire-pulse

![image](https://github.com/SteamGridDB/steam-rom-manager/assets/33595410/2cd4fe36-9bba-4864-9cf4-cc4445d1762e)

**Comment by erkexzcx** [comment]
+1

It freezes. I managed to add 6K+ ROMs to Steam Deck, but removing them using "(**Remove**) all added app entries and controllers" simply freezes the app and does not remove anything...

**EDIT**: As a workaround, I went to Linux Desktop on SteamDeck, opened Steam app, with mouse&keyboard I've selected all non-steam games (click on the top one, then while holding SHIFT click on the last one), then right click -> _Manage_ -> _Remove selected non-steam games from Steam_. Steam itself allows to bulk-remove non-steam games. Great success, but wish Steam ROM Manager could do it.
